### ISSUE TITLE
validate_arguments on the simplest func raise a NameError attributes shadowing, is this expected?

### ISSUE THREAD

**Issue by WestXu** [samuelcolvin/pydantic#1894]
```python
from pydantic import validate_arguments

@validate_arguments
def query(x: int, fields):
    pass
```
Run as-is, got a **NameError: Field name "fields" shadows a BaseModel attribute; use a different field name with "alias='fields'".**

Is this expected? Cause fields is a most frequently-used arg name. Noticed #1250, but validate_arguments even without type annotations (like `def query(fields)`) raises this error. I mean, not ideal. Any easy workaround on this one?

Cool project though.

**Comment by PrettyWood** [comment]
Hello @WestXu 
Unfortunately yes it's "normal" as it creates behind the scene a `BaseModel`
Feedback welcome on https://github.com/samuelcolvin/pydantic/issues/1205

**Comment by ezwelty** [comment]
Could this be addressed by internally using aliases for fields with colliding field names, then returning the validated values by their (original) alias names?

**Comment by zhqu1148980644** [comment]
same, any solutions?

**Comment by PrettyWood** [comment]
It should have been fixed in v1.8. @zhqu1148980644 which version are you using? If this is v1.8 please share a code snippet to reproduce

**Comment by zhqu1148980644** [comment]
@PrettyWood Thanks, it's v1.73, I'll try the newest version.
